File: MARVEL/Python/CorrectCodes/Jarvis.py
class MyString:
    def __init__(self, source):
        self.data = ""
        self.length = 0
        self.initialize_string(source)

    def initialize_string(self, source):
        self.length = len(source)
        self.data = source

    def print_string(self):
        for i in range(self.length):
            print(chr(ord(self.data[i]) - 1), end="")
        print()

    def free_string(self):
        self.data = ""
        self.length = 0


def main():
    my_string = MyString("NEVZMW")
    my_string.print_string()
    my_string.free_string()


if __name__ == "__main__":
    main()

File: MARVEL/C/correctCodes/Jarvis.c
#include <stdio.h>
#include <stdlib.h>

typedef struct {
    char *data;
    int length;
} MyString;

void initializeString(MyString *str, const char *source) {
    str->length = 0;

    while (source[str->length] != '\0') {
        str->length++;
    }

    str->data = (char *)malloc((str->length + 1) * sizeof(char));

    for (int i = 0; i < str->length; i++) {
        str->data[i] = source[i];
    }

    str->data[str->length] = '\0';
}

void printString(const MyString *str) {
    for (int i = 0; i < str->length; i++) {
        printf("%c", (str->data[i])-1);
    }
    printf("\n");
}


void freeString(MyString *str) {
    free(str->data);
    str->length = 0;
}

int main() {
    
    MyString myString;
    initializeString(&myString, "NEVZMW");
    printString(&myString);
    freeString(&myString);

    return 0;
}
